### PR TITLE
terraform 0.15.x support

### PR DESCRIPTION
**Pull request by RLRabinowitz**
Support terraform 0.15.x in terratag:

terraform 0.15.x has deprecated `map` in favor of `tomap`.

Solution:
- [x] Conditionally use `tomap` instead of `map`, if terraform version is higher than (or equal to) 0.15.0
- [x] Fix tests for terraform version 15 to have `tomap` instead of `map`

### CODE REVIEW

File: convert/convert.go [hunk 1]
```diff
@@ -49,7 +49,12 @@ func trimTokens(tokens hclwrite.Tokens) hclwrite.Tokens {
 	return tokens[startIndex : len(tokens)-endIndex]
 }
 
-func buildMapExpression(tokens hclwrite.Tokens) string {
+func buildMapExpression(tokens hclwrite.Tokens, tfVersion int) string {
+	if tfVersion >= 15 {
```
Comment (omry-hay) [inline convert/convert.go:53]: Since `tomap` was introduced in version 0.12 maybe we should change that.
```suggestion
	if tfVersion >= 12 {
```

**Review by omry-hay — APPROVED** [review]
Small comment.

File: convert/convert.go (hunk 1 above)
Comment (roni-frantchi) [reply convert/convert.go:53]: I wouldn't change older version generation at this point.

File: convert/convert.go [hunk 2]
```diff
@@ -13,9 +13,9 @@ import (
 	"strings"
 )
 
-func GetExistingTagsExpression(tokens hclwrite.Tokens) string {
+func GetExistingTagsExpression(tokens hclwrite.Tokens, tfVersion int) string {
 	if isHclMap(tokens) {
-		return buildMapExpression(tokens)
+		return buildMapExpression(tokens, tfVersion)
```
Comment (avnerenv0) [inline convert/convert.go:18]: BTW https://github.com/env0/terratag/issues/79
maybe would have been worth just doing that 🤷

Comment (RLRabinowitz) [reply convert/convert.go:18]: I think you're right. I was sure that `tomap` was needed here for the `merge`, but it isn't.
We can remove this buildMapExpression altogether probably, but that should only happen on a major version of terratag

**Pull request merged**